api/files/views.py

```diff
@@ -35,7 +35,235 @@ def get_file(self, check_permissions=True):
 
 
 class FileDetail(generics.RetrieveUpdateAPIView, FileMixin):
-    """Details about a specific file.
+    """Details about files and folders. *Writeable*.
+
+    Welcome to the Files API.  Brace yourself, things are about to get *weird*.
```

Comment:
Requests will still be weird, though. I'm good with that; I'm not a huge fan of the json api request format. We may optionally try to support that via waterbutler at some point?